internals/maintenance_scripts.py

```diff
@@ -564,14 +566,14 @@ def get_template_data(self, **kwargs):
 
     # OT stages that are flagged to process a trial creation.
     ot_stages: list[Stage] = Stage.query(
-        Stage.stage_type.IN(ALL_ORIGIN_TRIAL_STAGE_TYPES),
-        Stage.ot_action_requested == True).fetch()
+        Stage.ot_setup_status == OT_READY_FOR_CREATION).fetch()
```

Comment:
This will cause an issue if prepare_for_activation (or the trial activation) fails, but the OT trial is already marked as  OT_CREATED. I think we need a more robust error handling logic when the activation fails as well.

Questions: would this affect the new stage, e.g. stage2, that is linked to the previous OT? E.g. Different stages with the same OT?

Reply:
It's true that we need to still set the activation date on the stage if the activation initially fails. I added some extra enums for this situation, namely `OT_ACTIVATED` and `OT_ACTIVATION_FAILED`, which now capture the activation process as well. I also switched the enum values around to organize them correctly, which should be fine since this enum is not yet set for any stage.

My thought was that we can use `ot_activation_date` to understand where in the process the stage is, but the extra enum values seem more readable. 🙂 

> Questions: would this affect the new stage, e.g. stage2, that is linked to the previous OT? E.g. Different stages with the same OT?

I may be misunderstanding the question, but any origin trial stages should not be associated with more than 1 actual origin trial, if that's what you mean

Reply:
> My question was mostly related to the previous issue where two stages are linked to the same trial [1]. I guess it still requires us to fix it manually?

Ah yeah, that should not be an issue with the creation process, and exactly 1 trial will be associated with 1 trial stage